docs/paper/dev/api/commands-api/arguments.mdx

```diff
@@ -0,0 +1,190 @@
+---
+slug: /dev/commands-api/arguments
+description: A guide to Paper's Brigadier command API.
+---
+
+# Arguments
+
+Argumenttypes are datatypes that we can use instead of strings.
+
+:::danger[Experimental]
+
+Paper's command system is still experimental and may change in the future.
+
+:::
+
+## Basic usage of arguments
+
+You can add arguments to a command by doing the following:
+```java title="YourPluginClass.java"
+public class YourPluginClass extends JavaPlugin {
+    @Override
+    public void onEnable() {
+        LifecycleEventManager<Plugin> manager = this.getLifecycleManager();
+        manager.registerEventHandler(LifecycleEvents.COMMANDS, event -> {
+            final Commands commands = event.registrar();
+            commands.register(
+                Commands.literal("enchantmentargumentcommand")
+                .then(
+                    Commands.argument("enchantmentargument", ArgumentTypes.resource(RegistryKey.ENCHANTMENT))
+                        .executes(ctx -> {
+                            ctx.getSource().getSender().sendPlainMessage(
+                                ctx.getArgument("enchantmentargument", Enchantment.class).getKey().asString()
+                            );
+                            return Command.SINGLE_SUCCESS;
+                        })
+                ).build()
+            );
+        });
+    }
+}
+```
+This command has one argument of the `Enchantment` datatype. The advantage of using
+Brigadier over the `TabExecutor` interface is that you don't have to convert the
+string that is passed to the `CommandExecutor#onCommand()` function as an argument to 
+an enchantment in this case because `ctx.getArgument("enchantmentargument", Enchantment.class)`
+returns a value of the `Enchantment` datatype. Additionally, you don't have to write 
+error handling functionality yourself as the client checks whether the argument the
+user wrote in the chat is one of the `minecraft:enchantment` type and if it isn't,
+the user will get an error. Another advantage `ArgumentTypes` have over string-based argument systems like
+Bukkit's `TabExecutor` is that we can work with our own types and the completions list 
+is not always sorted alphanumerically. This may sound insignificant but when you want an input 
+of the type integer and suggest the numbers 0-10, for example, your list will look like this because
+it needs to be converted to strings:`0, 1, 10, 2, 3, ...`. Looks awful, right?
```

Comment:
this paragraph is quite complex, the long sentences could be broken up into something more easily digestible